src/IW/Effects/Cabal.hs

```diff
@@ -32,26 +33,31 @@ instance MonadCabal App where
 
 type WithCabal env m = (MonadDownload m, WithLog env m, WithError m)
 
-{- | This function may throw anyone of the errors inherited by the use of @downloadFile@
-defined in @IW.Effects.Download@. We are using @parseGenericPackageDescriptionMaybe@
-which will return @Nothing@ on an unsuccessful parse.
+{- | This function takes a @Repo@ and either returns @[Category]@, or
+throws a @CabalParseError@ error. This function may also throw any one of the
+errors inherited by the use of @downloadFile@ defined in @IW.Effects.Download@.
 -}
 getCabalCategoriesImpl
     :: forall env m.
        WithCabal env m
     => Repo
     -> m [Category]
 getCabalCategoriesImpl Repo{..} = do
-    maybeCabalFile <- downloadFileMaybe repoCabalUrl
-    case maybeCabalFile >>= parseGenericPackageDescriptionMaybe of
-        Nothing -> do
-            log W $ "Couldn't parse file downloaded from " <> unUrl repoCabalUrl
-            pure []
-        Just genPkgDescr -> do
-            log I $ "Successfully parsed file downloaded from " <> unUrl repoCabalUrl
+    cabalFile <- downloadFile repoCabalUrl
+    let result = runParseResult $ parseGenericPackageDescription cabalFile
```

Comment:
That would be helpful if you pattern match on the tuple here, like:

```haskell
let (warnings, somethingElse) = ...
```
if you add the comment about what this result is it will do as well :slightly_smiling_face:

Reply:
@vrom911 Oooh yes, I can do that. Much nicer.